dvc/repo/metrics/diff.py

```diff
@@ -12,6 +12,14 @@ def diff(repo, *args, a_rev=None, b_rev=None, **kwargs):
     b_rev = b_rev or "workspace"
 
     metrics = repo.metrics.show(*args, **kwargs, revs=[a_rev, b_rev])
+
+    # workspace may have been replaced by active branch
+    workspace_rev = (a_rev == "workspace") or (b_rev == "workspace")
+    if workspace_rev and "workspace" not in metrics:
+        active_branch = repo.scm.active_branch()
+        if active_branch in metrics:
+            metrics["workspace"] = metrics[active_branch]
+
```

Comment:
Does seem pretty weird that `show` doesn't return it at all. Looks like https://github.com/iterative/dvc/blob/0e5674ffeabb4f911a9486da8ff3ba3a33d1d98c/dvc/repo/metrics/show.py#L145 is the `metrics show`-specific behaviour that leaked into `metrics diff`. Probably needs a flag not to hide workspace metrics or it should hide it in CLI (that would be my expectation). Taking another look...

Reply:
@dberenbaum @daavoo Guys, do you find this https://github.com/iterative/dvc/blob/0e5674ffeabb4f911a9486da8ff3ba3a33d1d98c/dvc/repo/metrics/show.py#L145 behaviour useful at all in the API? I'm kinda tempted to move that hiding logic to the CLI layer so we don't have to fix it in `diff` at all. If having that in API is useful, then I'll make it an opt-out flag, so again we don't have to fix it in `diff`.

Reply:
Added a flag for now.